fastcrypto/src/tests/bls12381_group_tests.rs

```diff
@@ -653,3 +656,64 @@ fn test_serialization_gt() {
     assert!(GTElement::from_trusted_byte_array(&bytes).is_ok());
     assert!(GTElement::from_byte_array(&bytes).is_err());
 }
+
+#[test]
+fn test_g1_to_uncompressed() {
+    let a = G1Element::generator() * Scalar::from(7u128);
+
+    let uncompressed_bytes = G1ElementUncompressed::from(&a);
+
+    // Compressed bit flags (1 and 3) should not be set.
+    assert_eq!(uncompressed_bytes.0[0] & 0xA0, 0);
+
+    // Infinity bit flag (2) should not be set.
+    assert_eq!(uncompressed_bytes.0[0] & 0x40, 0);
+
+    // Regression test
+    assert_eq!(&uncompressed_bytes.0, hex::decode("1928f3beb93519eecf0145da903b40a4c97dca00b21f12ac0df3be9116ef2ef27b2ae6bcd4c5bc2d54ef5a70627efcb7108dadbaa4b636445639d5ae3089b3c43a8a1d47818edd1839d7383959a41c10fdc66849cfa1b08c5a11ec7e28981a1c").unwrap().as_slice());
+
+    // Serialize the point-at-infinity
+    let a = G1Element::zero();
+    let uncompressed_bytes = G1ElementUncompressed::from(&a);
+
+    // Only the point at infinity flag should be set.
+    assert_eq!(uncompressed_bytes.0[0], 0x40);
+
+    // The remaining bytes should all be zero
+    assert_eq!(
+        uncompressed_bytes.0[1..],
+        [0u8; G1_ELEMENT_BYTE_LENGTH * 2 - 1]
+    );
+}
+
+#[test]
+fn test_g1_sum() {
+    // Empty sum
+    assert_eq!(G1ElementUncompressed::sum(&[]).unwrap(), G1Element::zero());
+
+    // Non-trivial sum
+    let a = G1Element::generator();
+    let b = G1Element::generator() * Scalar::from(2u128);
+    let c = G1Element::generator() * Scalar::from(3u128);
+    let mut bytes: Vec<G1ElementUncompressed> = vec![(&a).into(), (&b).into(), (&c).into()];
+    let sum = G1ElementUncompressed::sum(&bytes).unwrap();
+    assert_eq!(sum, G1Element::generator() * Scalar::from(6u128));
+
+    // Adding zeros doesn't change anything
+    bytes.push(G1ElementUncompressed::from(&G1Element::zero()));
+    let sum = G1ElementUncompressed::sum(&bytes).unwrap();
+    assert_eq!(sum, G1Element::generator() * Scalar::from(6u128));
+
+    // Singleton sum
+    let bytes = [(&b).into()];
+    let sum = G1ElementUncompressed::sum(&bytes).unwrap();
+    assert_eq!(sum, b);
+
+    // Adding zero's
+    let mut bytes = vec![G1ElementUncompressed::from(&G1Element::zero())];
+    let sum = G1ElementUncompressed::sum(&bytes).unwrap();
+    assert_eq!(sum, G1Element::zero());
+    bytes.push(G1ElementUncompressed::from(&G1Element::zero()));
+    let sum = G1ElementUncompressed::sum(&bytes).unwrap();
+    assert_eq!(sum, G1Element::zero());
+}
```

Comment:
also let's add a short test that: chooses 100 random scalars -> computes sum using compressed vs uncompressed -> compares